token/read.go

```diff
@@ -15,12 +16,55 @@ import (
 	"github.com/ucan-wg/go-ucan/token/invocation"
 )
 
+// FromSealed decodes an arbitrary token type from the binary data,
+// verifies that the envelope's signature is correct based on the public
+// key taken from the issuer (iss) field and calculates the CID of the
+// incoming data.
+// Supported and returned types are:
+// - delegation.Token
+// - invocation.Token
+func FromSealed(data []byte) (Token, cid.Cid, error) {
+	tkn, err := FromDagCbor(data)
+	if err != nil {
+		return nil, cid.Undef, err
+	}
+
+	id, err := envelope.CIDFromBytes(data)
+	if err != nil {
+		return nil, cid.Undef, err
+	}
+
+	// tkn.cid = id
```

Comment:
This is where the problem is: no way to do that without breaking encapsulation.
My thinking is that we don't actually need to have the CID in the token, it seems enough to carry it externally (aka, in a container).